document deployments

nicksnyder opened the pull request:
This PR adds information to our handbook about deployments. I think this is better documented here in our handbook than in deploy-sourcegraph-dot-com because the former is more accessible and discoverable than the latter.

https://github.com/sourcegraph/deploy-sourcegraph-dot-com/pull/1861 deletes duplicate information from that repository.

ggilmore commented:
Can we link to https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/README.info.md, and/or de-duplicate this information?

nicksnyder commented:
@ggilmore Yep, good idea! Even I forgot about that readme 😬. Done.

tsenart commented:
This doesn't document how the deploys happen, where to check status, etc. Is that coming in a different PR?

nicksnyder commented:
> This doesn't document how the deploys happen, where to check status, etc. Is that coming in a different PR?

I hadn't planned any other PRs. Does the documentation in https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/README.info.md#deploy-your-changes-to-sourcegraphcom (linked to by this document) not answer your question? If not, does [my previous wording](https://github.com/sourcegraph/about/pull/487/commits/64de32c069f5453d1f25934a07aa087a1ddc7bb1) answer your question?

The status of a deployment is captured in the CI step that runs the `kubectl apply`.

tsenart commented:
@nicksnyder: I think putting all of this information together in this handbook page is much better than scattering it, so I'm for your previous approach. I'd like the same for other environments though, not only dotcom.

As for deployment status visibility, it's not captured by the CI step because of the whole async Renovate workflow. As an engineer, I can't see this in a single place, and I don't know how long I'd have to wait until my change is deployed.

nicksnyder commented:
>  I think putting all of this information together in this handbook page is much better than scattering it, so I'm for your previous approach.

After thinking about it some more I think I agree, so I switched it back.

>  I'd like the same for other environments though, not only dotcom.

The other environments are the same as dotcom, just the configuration is different. I added a note to clarify.

> As for deployment status visibility, it's not captured by the CI step because of the whole async Renovate workflow. As an engineer, I can't see this in a single place, and I don't know how long I'd have to wait until my change is deployed.

Can you read the updated docs I wrote to see if that answers your questions? Unfortunately there is not a single place to look because out deploy process involves multiple systems right now.

tsenart commented on handbook/engineering/deployments.md:
```diff
@@ -0,0 +1,51 @@
+
+Every commit to the `release` branch (the default branch) on [deploy-sourcegraph-dot-com](https://github.com/sourcegraph/deploy-sourcegraph-dot-com) deploys the Kubernetes YAML in this repository to our dot-com cluster [in CI](https://buildkite.com/sourcegraph/deploy-sourcegraph-dot-com/builds?branch=release) (i.e. if CI is green then the latest config in the `release` branch is deployed).
+
+If you want to expedite a deploy, you can manually create and merge a PR that updates the Docker image tags in [sourcegraph/deploy-sourcegraph-dot-com](https://github.com/sourcegraph/deploy-sourcegraph-dot-com). You can find the desired Docker image tags by looking at the output of the Docker build step in [CI on sourcegraph/sourcegraph `master` branch](https://buildkite.com/sourcegraph/sourcegra ...
```
Does anyone have a script around to do this manual step? cc @ggilmore? If so, we should document it here.

tsenart commented:
Thanks for this @nicksnyder. You should post it to #dev-chat once it's live.

nicksnyder commented on handbook/engineering/deployments.md:
```diff
@@ -0,0 +1,51 @@
+We maintain multiple deployments of Sourcegraph:
+
+- sourcegraph.com is our production deployment for open source code.
+  - [dot-com Kubernetes cluster on GCP](https://console.cloud.google.com/kubernetes/clusters/details/us-central1-f/dot-com?project=sourcegraph-dev)
```
```suggestion
  - [dot-com cluster on GCP](https://console.cloud.google.com/kubernetes/clusters/details/us-central1-f/dot-com?project=sourcegraph-dev)
```

nicksnyder commented on handbook/engineering/deployments.md:
```diff
@@ -0,0 +1,51 @@
+
+Every commit to the `release` branch (the default branch) on [deploy-sourcegraph-dot-com](https://github.com/sourcegraph/deploy-sourcegraph-dot-com) deploys the Kubernetes YAML in this repository to our dot-com cluster [in CI](https://buildkite.com/sourcegraph/deploy-sourcegraph-dot-com/builds?branch=release) (i.e. if CI is green then the latest config in the `release` branch is deployed).
+
+If you want to expedite a deploy, you can manually create and merge a PR that updates the Docker image tags in [sourcegraph/deploy-sourcegraph-dot-com](https://github.com/sourcegraph/deploy-sourcegraph-dot-com). You can find the desired Docker image tags by looking at the output of the Docker build step in [CI on sourcegraph/sourcegraph `master` branch](https://buildkite.com/sourcegraph/sourcegra ...
```
```suggestion
If you want to expedite a deploy, you can manually create and merge a PR that updates the Docker image tags in [deploy-sourcegraph-dot-com](https://github.com/sourcegraph/deploy-sourcegraph-dot-com). You can find the desired Docker image tags by looking at the output of the Docker build step in [CI on sourcegraph/sourcegraph `master` branch](https://buildkite.com/sourcegraph/sourcegraph/builds?branch=master) or by looking at [Docker Hub](https://hub.docker.com/u/sourcegraph/).
```
